File: fileuploadanddownload/src/main/java/qzl/com/fileuploadanddownload/popuWindow/ExpandablePopupWindow.java
package qzl.com.fileuploadanddownload.popuWindow;

import android.app.Activity;
import android.content.Context;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.*;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import qzl.com.basecommon.base.BasCommonPopuWindow;
import qzl.com.basecommon.base.BasePopuWindow;
import qzl.com.fileuploadanddownload.R;
import qzl.com.tools.utils.ScreenUtil;

import java.util.List;

public class ExpandablePopupWindow extends BasCommonPopuWindow {
    private ListView popupListView;
    private Activity mActivity;
    private List<String> list;
    private OnSelectedItemListener onSelectedItemListener;
    private View mView;
    public ExpandablePopupWindow(Activity activity,View view, List<String> list,OnSelectedItemListener onSelectedItemListener){
        super(activity);
        this.mActivity = activity;
        this.list = list;
        this.mView = view;
        this.onSelectedItemListener = onSelectedItemListener;
        setPopWindowContent(R.layout.pop_material);
    }
    @Override
    protected void showPopuwindow(@NotNull BasePopuWindow popuWindow) {
        popuWindow.showAtLocation(mView, Gravity.BOTTOM, 0, ScreenUtil.getVirtualBarHeight(mActivity));
    }

    @Override
    protected void initComplate(@NotNull PopupWindow popupWindow, @Nullable View pupView) {
        popupListView = pupView.findViewById(R.id.pop_item_list);
        popupListView.setAdapter(new PopupWindowAdapter(mActivity, list));
        popupListView.setOnItemClickListener(onItemClickListener);
    }

    public interface OnSelectedItemListener{
        abstract void onSelectedValue(String str);
    }

    private AdapterView.OnItemClickListener onItemClickListener = new AdapterView.OnItemClickListener() {
        @Override
        public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
            onSelectedItemListener.onSelectedValue(parent.getAdapter().getItem(position).toString());
            dismiss();
        }
    };

    private class PopupWindowAdapter extends BaseAdapter{

        private Context context;
        private List<String> list;

        public PopupWindowAdapter(Context context, List<String> list) {
            this.context = context;
            this.list = list;
        }

        @Override
        public int getCount() {
            if (list != null && !list.isEmpty())
                return list.size();
            return 0;
        }

        @Override
        public Object getItem(int position) {
            return list.get(position);
        }

        @Override
        public long getItemId(int position) {
            return position;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            if (convertView == null) {
                convertView = LayoutInflater.from(context).inflate(R.layout.i_pop_material_type, parent, false);
            }
            TextView popMaterialType = (TextView) convertView.findViewById(R.id.pop_material_type);
            popMaterialType.setText(list.get(position));
            return convertView;
        }
    }
}

File: QGMultipleModelFrame/base_common/src/main/java/qzl/com/basecommon/popuwindow/ExpandablePopupWindow.kt
package qzl.com.basecommon.popuwindow

import android.content.Context
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ListView
import android.widget.PopupWindow
import android.widget.TextView
import qzl.com.basecommon.R
import qzl.com.basecommon.base.BasCommonPopuWindow
import qzl.com.basecommon.base.BasePopuWindow
import qzl.com.tools.utils.ScreenUtil.Companion.getVirtualBarHeight

class ExpandablePopupWindow(val activity: Context, val list: List<String>,val mView:View) : BasCommonPopuWindow(activity) {
    private var onSelectedItemListener: OnSelectedItemListener? = null

    init {
        setPopWindowContent(R.layout.pop_event_material)
    }
    override fun showPopuwindow(popuWindow: BasePopuWindow) {
        popuWindow.showAtLocation(mView, Gravity.BOTTOM, 0, getVirtualBarHeight(activity))
    }

    override fun initComplate(popupWindow: PopupWindow, pupView: View?) {
        val listView = pupView?.findViewById<ListView>(R.id.pop_item_list)
        listView?.adapter = PopupWindowAdapter(activity, list)
        listView?.setOnItemClickListener { parent, view, position, id ->
            onSelectedItemListener?.onSelectedValue(parent.adapter.getItem(position).toString(),position)
            popupWindow.dismiss()
        }
    }

    interface OnSelectedItemListener {
        fun onSelectedValue(str: String,position: Int)
    }

    fun setOnSelectedItemListener(onSelectedItemListener: OnSelectedItemListener?) {
        this.onSelectedItemListener = onSelectedItemListener
    }

    private inner class PopupWindowAdapter(private val context: Context, private val list: List<String>) : BaseAdapter() {
        override fun getCount(): Int {
            return if (list.isNotEmpty()) {
                list.size
            } else 0
        }

        override fun getItem(position: Int): Any {
            return list[position]
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
            var convertView = convertView
            if (convertView == null) {
                convertView = LayoutInflater.from(context).inflate(R.layout.i_pop_material_type, parent, false)
            }
            val popMaterialType = convertView?.findViewById<TextView>(R.id.pop_material_type)
            popMaterialType?.text = list[position]
            return convertView
        }
    }
}